extensions/wikia/FirstContributions/api/ApiQueryFirstContributions.php

```diff
@@ -0,0 +1,124 @@
+<?php
+
+/**
+ * API module which returns list of users' first edits, including revision id, date and user name
+ * @ingroup API
+ */
+class ApiQueryFirstContributions extends ApiQueryBase {
+
+	public function __construct( ApiBase $query, $moduleName ) {
+		parent::__construct( $query, $moduleName, 'fc' );
+	}
+
+	public function execute() {
+		$params = $this->extractRequestParams();
+
+		$this->addTables( 'recentchanges' );
+
+		$this->addFields( 'rc_this_oldid as id' );
+		$this->addFields( 'rc_timestamp as timestamp' );
+		$this->addFields( 'rc_user_text as author' );
+
+		$this->addWhere( 'rc_type < ' . RC_MOVE );
+		$this->addWhere( [ 'rc_bot' => 0 ] );
```

Comment:
Please paste here an example SQL query that will be made by this code + the EXPLAIN